File: packages/command-client/src/tauri/secret-store.ts
import { call } from "./invoke";
import type {
  CredentialCreateInput,
  CredentialDeleteInput,
  CredentialInspectInput,
  CredentialMetadata,
  CredentialRotateInput,
} from "../types";

export function createCredential(input: CredentialCreateInput) {
  return call<CredentialMetadata>("credential_create", { input });
}

export function deleteCredential(input: CredentialDeleteInput) {
  return call<void>("credential_delete", { input });
}

export function inspectCredential(input: CredentialInspectInput) {
  return call<CredentialMetadata>("credential_inspect", { input });
}

export function rotateCredential(input: CredentialRotateInput) {
  return call<CredentialMetadata>("credential_rotate", { input });
}

File: crates/unfour-app/src/commands/secret_store.rs
use crate::AppState;
use tauri::State;
use unfour_core::{
    models::{
        CredentialCreateInput, CredentialDeleteInput, CredentialInspectInput, CredentialMetadata,
        CredentialRotateInput,
    },
    AppResult,
};

use super::trace_command;

#[tauri::command]
pub async fn credential_create(
    input: CredentialCreateInput,
    state: State<'_, AppState>,
) -> AppResult<CredentialMetadata> {
    trace_command(
        "credential_create",
        state.command_bus.create_credential(input),
    )
    .await
}

#[tauri::command]
pub async fn credential_delete(
    input: CredentialDeleteInput,
    state: State<'_, AppState>,
) -> AppResult<()> {
    trace_command(
        "credential_delete",
        state.command_bus.delete_credential(input),
    )
    .await
}

#[tauri::command]
pub async fn credential_inspect(
    input: CredentialInspectInput,
    state: State<'_, AppState>,
) -> AppResult<CredentialMetadata> {
    trace_command(
        "credential_inspect",
        state.command_bus.inspect_credential(input),
    )
    .await
}

#[tauri::command]
pub async fn credential_rotate(
    input: CredentialRotateInput,
    state: State<'_, AppState>,
) -> AppResult<CredentialMetadata> {
    trace_command(
        "credential_rotate",
        state.command_bus.rotate_credential(input),
    )
    .await
}
